lbrynet/core/client/ConnectionManager.py

```diff
@@ -137,37 +142,21 @@ def _send_secondary_requests(self, peer, protocol):
         ]
         return defer.DeferredList(ds)
 
-    def protocol_disconnected(self, peer, protocol):
-        log.debug("%s protocol %s disconnected for %s",
-                    self._get_log_name(), type(protocol).__name__, peer)
-        if peer in self._peer_connections:
-            del self._peer_connections[peer]
-        if peer in self._connections_closing:
-            d = self._connections_closing[peer]
-            del self._connections_closing[peer]
-            d.callback(True)
-
     @defer.inlineCallbacks
-    def _manage(self):
+    def manage(self, schedule_next_call=True):
         self._manage_deferred = defer.Deferred()
-
-        from twisted.internet import reactor
         if len(self._peer_connections) < conf.settings['max_connections_per_stream']:
             log.debug("%s have %d connections, looking for %d",
                         self._get_log_name(), len(self._peer_connections),
                         conf.settings['max_connections_per_stream'])
-            try:
-                ordered_request_creators = self._rank_request_creator_connections()
-                peers = yield self._get_new_peers(ordered_request_creators)
-                peer = self._pick_best_peer(peers)
-                yield self._connect_to_peer(peer)
-            except Exception:
-                # log this otherwise it will just end up as an unhandled error in deferred
-                log.exception('Something bad happened picking a peer')
+            ordered_request_creators = self._rank_request_creator_connections()
+            peers = yield self._get_new_peers(ordered_request_creators)
+            peer = self._pick_best_peer(peers)
+            self._connect_to_peer(peer)
```

Comment:
I believe this exception catching is no longer needed. When connection fails (replicated in non existing server unit test), ClientProtocolFactory.clientConnectionFailed() is called. Before there was no logging information placed in clientConnectionFailed (added in https://github.com/lbryio/lbry/pull/441),  so the exception here would handle the catching and logging.   Now that we have 441, we don't need to catch and log here.